tests/analyzer/test_rulebase.py

```diff
@@ -46,10 +46,8 @@ def test_rule_base(self):
         )
 
         # Test - _preprocess
-        rules = rulebase1._preprocess(test_raw_data_fake, test_rule_file)
-        assert (not rules)
-        rules = rulebase1._preprocess(test_raw_data, test_rule_file_fake)
-        assert (not rules)
+        self.assertRaises(Exception, rulebase1._preprocess, test_raw_data_fake, test_rule_file)
+        self.assertRaises(Exception, rulebase1._preprocess, test_raw_data, test_rule_file_fake)
```

Comment:
do you need to compare the actual exception instead of the base class? otherwise the test seems useless if the error changes from exceptionA to exceptionB..

Reply:
thanks, updated in another pr